x/slashing/proposal_handler_test.go

```diff
@@ -0,0 +1,63 @@
+package slashing_test
+
+import (
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/slashing"
+	"github.com/cosmos/cosmos-sdk/x/slashing/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/teststaking"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"testing"
+)
+
+func TestSlashValidatorProposal(t *testing.T) {
+	// initial setup
```

Comment:
Can we add another comment for whats the setup logic / what its testing?

Appears to me: Setup a staking module with max 1 validator,  and then two validators in queue. Run a block. Slash the validator thats not in the active set via proposal (it fails)

Slash the validator in the active set (it succeeds)

What I don't get / should we be testing: Does the #2 validator then become the #1 validator and the first validator is jailed, or is this not jailing?

Reply:
yeah, in general i spent a lot of time reading slashing tests and trying understand them myself.
Slashing module itself is nightmare from my point of view, lots of panic and other staff.
Here is copy-paste from test slash module. In general i don't understand why there are not test setups helpers that can be used in other modules. But osmo fork is not that place, where it should be fixed :(

Will add more comments here about our test, to make it clear whats going on.

It is not jailing, only slashing. Should we consider jailing as proposal option?

Reply:
probably, there could be circumstances where a validator needs to be removed from the active set.

Reply:
Oh yikes, didn't realize the slashing tests weren't well architected to begin with :/